kolibri/core/assets/src/views/icon-button/index.vue

```diff
@@ -1,85 +1,63 @@
 <template>
 
-  <ui-button
-    @click="$emit('click')"
-    :color="primary ? 'primary' : 'default'"
+  <button
+    class="k-button"
+    ref="button"
+    :class="classes"
+    :type="type"
     :disabled="disabled"
-    :buttonType="type"
-    :size="size"
-    :loading="loading"
-    class="koli-icon-button">
-    <span v-if="hasIcon && text && alignLeft">
-      <ui-icon class="icon-left"><slot/></ui-icon>
-      {{ text }}
-    </span>
-    <span v-else-if="hasIcon && text && alignRight">
-      {{ text }}
-      <ui-icon class="icon-right"><slot/></ui-icon>
-    </span>
-    <span v-else-if="hasIcon">
-      <ui-icon><slot/></ui-icon>
-    </span>
-    <span v-else>
-      {{ text }}
-    </span>
-  </ui-button>
+    @click="handleClick">
+    {{ text }}
+  </button>
 
 </template>
 
 
 <script>
 
-  import uiButton from 'keen-ui/src/UiButton';
-  import uiIcon from 'keen-ui/src/UiIcon';
   export default {
     props: {
-      text: { type: String },
+      text: {
+        type: String,
+        required: true,
+      },
       primary: {
         type: Boolean,
         default: false,
       },
-      disabled: {
+      raised: {
         type: Boolean,
-        default: false,
+        default: true,
       },
-      showTextBelowIcon: {
+      disabled: {
         type: Boolean,
         default: false,
       },
-      size: {
-        type: String,
-        default: 'normal',
-      },
       type: { type: String },
-      alignment: {
-        type: String,
-        default: 'left',
-        required: false,
-        validator(value) {
-          return value === 'left' || value === 'right';
-        },
-      },
-      loading: {
-        type: Boolean,
-        required: false,
-      },
     },
     computed: {
-      hasIcon() {
-        // check if the parent passed anything into the slot
-        // $slots returns an empty object if nothing is passed in.
-        return !(Object.keys(this.$slots).length === 0 && this.$slots.constructor === Object);
-      },
-      alignLeft() {
-        return this.alignment === 'left';
-      },
-      alignRight() {
-        return this.alignment === 'right';
+      classes() {
+        let classes = [];
+        if (this.primary && this.raised) {
+          classes.push('k-button-primary-raised');
+        } else if (this.primary && !this.raised) {
+          classes.push('k-button-primary-flat');
+        } else if (!this.primary && this.raised) {
+          classes.push('k-button-secondary-raised');
+        } else if (!this.primary && !this.raised) {
+          classes.push('k-button-secondary-flat');
+        }
+        if (this.disabled) {
+          classes.push('k-button-disabled');
+        }
+        return classes;
       },
     },
-    components: {
-      uiButton,
-      uiIcon,
+    methods: {
+      handleClick(event) {
+        this.$emit('click', event);
+        this.$refs.button.blur();
```

Comment:
Right - it's subtle but perhaps [not quite correct](https://marcysutton.com/button-focus-hell/)?

> clicking sets focus, but not the visible focus style. Hitting TAB will show the focus style on the next button, since it was an intentional keyboard interaction

That page includes some rather subtle but straightforward timer-based logic for making this work across browsers.

I think your solution is fine for a first-pass, but just flagging that we may want to tweak the behavior eventually.